plugins/modules/docker_compose_v2.py

```diff
@@ -0,0 +1,926 @@
+        project_src: flask
+        state: absent
+
+    - community.docker.docker_compose:
+        project_name: flask
+        definition:
+          version: '2'
+          services:
+            db:
+              image: postgres
+            web:
+              build: "{{ playbook_dir }}/flask"
+              command: "python manage.py runserver 0.0.0.0:8000"
+              volumes:
+                - "{{ playbook_dir }}/flask:/code"
+              ports:
+                - "8000:8000"
+              depends_on:
+                - db
+      register: output
+
+    - ansible.builtin.debug:
+        var: output
+
+    - ansible.builtin.assert:
+        that:
+          - "'started' in containers['flask_web_1'] | default([])"
+          - "'started' in containers['flask_db_1'] | default([])"
+
+- name: Run with inline Compose file version 1
+  # https://docs.docker.com/compose/compose-file/compose-file-v1/
+  hosts: localhost
+  gather_facts: false
+  tasks:
+    - community.docker.docker_compose:
+        project_src: flask
+        state: absent
+
+    - community.docker.docker_compose:
+        project_name: flask
+        definition:
+            db:
+              image: postgres
+            web:
+              build: "{{ playbook_dir }}/flask"
+              command: "python manage.py runserver 0.0.0.0:8000"
+              volumes:
+                - "{{ playbook_dir }}/flask:/code"
+              ports:
+                - "8000:8000"
+              links:
+                - db
+      register: output
+
+    - ansible.builtin.debug:
+        var: output
+
+    - ansible.builtin.assert:
+        that:
+          - "'started' in containers['flask_web_1'] | default([])"
+          - "'started' in containers['flask_db_1'] | default([])"
+'''
+
+RETURN = '''
+stdout:
+  description:
+    - The stdout from docker-compose.
+  returned: always, unless when C(docker-compose) was not given the chance to run
+  type: str
+stderr:
+  description:
+    - The stderr from docker-compose.
+  returned: always, unless when C(docker-compose) was not given the chance to run
+  type: str
+containers:
+  description:
+    - A dictionary mapping the various status of containers during C(docker-compose) operation.
+  returned: always, unless when C(docker-compose) was not given the chance to run
+  type: complex
+  contains:
+    container_name:
+      description: Name of the container.
+      returned: always, unless when C(docker-compose) was not given the chance to run
+      type: list
+      elements: str
+      example: ["stopped", "removed"]
+volumes:
+  description:
+    - A dictionary mapping the various status of volumes during C(docker-compose) operation.
+  returned: always, unless when C(docker-compose) was not given the chance to run
+  type: complex
+  contains:
+    volume_name:
+      description: Name of the volume.
+      returned: always, unless when C(docker-compose) was not given the chance to run
+      type: list
+      elements: str
```

Comment:
If this is a single name, why is it a list of strings?

Reply:
I think I misunderstand how dicts are built `RETURN`.

The elements `containers`, `images`, `volumes` and `networks` looks like this:
```json
{
  "name_of_the_resource": [
    "statuses",
    "it",
    "went",
    "through",
    "during",
    "run"
  ]
}
```

Reply:
Fixed, see https://github.com/ansible-collections/community.docker/pull/586#discussion_r1114211046